drivers/gpio/gpio_renesas_ra8.c

```diff
@@ -0,0 +1,234 @@
+			}
+		}
+
+		R_SYSTEM->PRCR = ((R_SYSTEM->PRCR | PRCR_KEY) | R_SYSTEM_PRCR_PRC1_Msk);
+
+		R_SYSTEM->VBTICTLR &= (uint8_t)~clear;
+
+		R_SYSTEM->PRCR = ((R_SYSTEM->PRCR | PRCR_KEY) & (uint16_t)~R_SYSTEM_PRCR_PRC1_Msk);
+	}
+
+	R_PMISC->PWPRS = 0;
+	R_PMISC->PWPRS = BIT(R_PMISC_PWPR_PFSWE_Pos);
+
+	uint32_t value = R_PFS->PORT[config->port_num].PIN[pin].PmnPFS;
+
+	/* Change mode to general IO mode and disable IRQ and Analog input */
+	WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_PMR_Pos, 0);
+	WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_ASEL_Pos, 0);
+	WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_ISEL_Pos, 0);
+
+	if ((flags & GPIO_OUTPUT) != 0U) {
+		/* Set output pin initial value */
+		if ((flags & GPIO_OUTPUT_INIT_LOW) != 0U) {
+			WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_PODR_Pos, 0);
+		} else if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0U) {
+			WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_PODR_Pos, 1);
+		}
+
+		WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_PDR_Pos, 1);
+	} else {
+		WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_PDR_Pos, 0);
+	}
+
+	R_PFS->PORT[config->port_num].PIN[pin].PmnPFS = value;
+
+	R_PMISC->PWPRS = 0;
+	R_PMISC->PWPRS = BIT(R_PMISC_PWPR_B0WI_Pos);
+
+	return 0;
+}
+
+static int gpio_ra8_port_get_raw(const struct device *dev, uint32_t *value)
+{
+	const struct gpio_ra8_config *config = dev->config;
+	R_PORT0_Type *port = config->port;
+
+	*value = port->PIDR;
+
+	return 0;
+}
+
+static int gpio_ra8_port_set_masked_raw(const struct device *dev, gpio_port_pins_t mask,
+					gpio_port_value_t value)
+{
+	const struct gpio_ra8_config *config = dev->config;
+	R_PORT0_Type *port = config->port;
+
+	port->PODR = ((port->PODR & ~mask) | (value & mask));
+
+	return 0;
+}
+
+static int gpio_ra8_port_set_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_ra8_config *config = dev->config;
+	R_PORT0_Type *port = config->port;
+
+	port->PODR = (port->PODR | pins);
+
+	return 0;
+}
+
+static int gpio_ra8_port_clear_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_ra8_config *config = dev->config;
+	R_PORT0_Type *port = config->port;
+
+	port->PODR = (port->PODR & ~pins);
+
+	return 0;
+}
+
+static int gpio_ra8_port_toggle_bits(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_ra8_config *config = dev->config;
+	R_PORT0_Type *port = config->port;
+
+	port->PODR = (port->PODR ^ pins);
+
+	return 0;
+}
+
+static const struct gpio_driver_api gpio_ra8_drv_api_funcs = {
+	.pin_configure = gpio_ra8_pin_configure,
+	.port_get_raw = gpio_ra8_port_get_raw,
+	.port_set_masked_raw = gpio_ra8_port_set_masked_raw,
+	.port_set_bits_raw = gpio_ra8_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_ra8_port_clear_bits_raw,
+	.port_toggle_bits = gpio_ra8_port_toggle_bits,
+	.pin_interrupt_configure = NULL,
+	.manage_callback = NULL,
+#ifdef CONFIG_GPIO_GET_DIRECTION
+	.port_get_direction = gpio_ra8_port_get_direction,
+#endif
+};
+
+static int gpio_ra8_init(const struct device *dev)
```

Comment:
`DEVICE_DT_DEFINE` accepts NULL as initializer function pointer.
This function can be removed.